Lab5/src/controllers/user/user.controller.ts

```diff
@@ -0,0 +1,90 @@
+import {UserService} from "../../services/services";
+import {Request, Response, NextFunction} from "express";
+import {validateOrReject} from "class-validator";
+import {plainToInstance} from "class-transformer";
+import {UserEntity} from "../../entity/user/user.entity";
+import {UserProperties} from "../../common/types/user/user.properties";
+import {UserValidator} from "../../entity/user/user-validator";
+
+class UserController {
+
+    private readonly userService: UserService;
+    private readonly userValidator: UserValidator;
+
+
+    constructor(userService: UserService, userValidator: UserValidator) {
+        this.userService = userService;
+        this.userValidator = userValidator;
+    }
+
+    get = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+        try {
+            const id: number = Number(req.params.id);
+            this.userValidator.isIdValid(id);
+            const user: UserEntity | undefined = await this.userService.get(id);
+            res.json(user);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    getAll = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+        try {
+            const users: UserEntity[] = await this.userService.getAll();
+            res.json(users);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    create = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+        try {
+            const userJson = req.body;
```

Comment:
you get id from request?

Reply:
No, in line below i check is id undefined in userDto

Reply:
yep, you get id from request... yes you validate that it exist, but... Don't you think that ID generation should be done by server, not by client?